cmd/src/debug_common.go

```diff
@@ -0,0 +1,470 @@
+	// TODO: It may be nice to store a list of pods for which --previous isn't collected, to be outputted with verbose flag
+	for _, pod := range pods.Items {
+		for _, container := range pod.Spec.Containers {
+			wg.Add(1)
+			go func(pod, container string) {
+				if err := semaphore.Acquire(ctx, 1); err != nil {
+					// return err
+				}
+				defer semaphore.Release(1)
+				defer wg.Done()
+				f := getPastContainerLog(ctx, pod, container, namespace, baseDir)
+				if f.err == nil {
+					ch <- f
+				} else {
+					if verbose {
+						fmt.Printf("Could not gather --previous pod logs for: %s \nExited with err: %s\n", pod, f.err)
+					}
+				}
+			}(pod.Metadata.Name, container.Name)
+		}
+	}
+
+	// start goroutine for each pod to run kubectl describe pod
+	for _, pod := range pods.Items {
+		wg.Add(1)
+		go func(pod string) {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				// return err
+			}
+			defer semaphore.Release(1)
+			defer wg.Done()
+			ch <- getDescribe(ctx, pod, namespace, baseDir)
+		}(pod.Metadata.Name)
+	}
+
+	// start goroutine for each pod to run kubectl get pod <pod> -o yaml
+	for _, pod := range pods.Items {
+		wg.Add(1)
+		go func(pod string) {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				// return err
+			}
+			defer semaphore.Release(1)
+			defer wg.Done()
+			ch <- getManifest(ctx, pod, namespace, baseDir)
+		}(pod.Metadata.Name)
+	}
+
+	// start goroutine to get external service config
+	if configs == true {
+		wg.Add(1)
+		go func() {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				// return err
+			}
+			defer semaphore.Release(1)
+			defer wg.Done()
+			ch <- getExternalServicesConfig(ctx, baseDir)
+		}()
+
+		wg.Add(1)
+		go func() {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				// return err
+			}
+			defer semaphore.Release(1)
+			defer wg.Done()
+			ch <- getSiteConfig(ctx, baseDir)
+		}()
+	}
+
+	// close channel when wait group goroutines have completed
+	go func() {
+		wg.Wait()
+		close(ch)
+	}()
+
+	// write to archive all the outputs from kubectl call functions passed to buffer channel
+	for f := range ch {
+		if f.err != nil {
+			log.Printf("getting data for %s failed: %v\noutput: %s", f.name, f.err, f.data)
+			continue
+		}
+
+		if verbose {
+			log.Printf("archiving file %q with %d bytes", f.name, len(f.data))
+		}
+
+		zf, err := zw.Create(f.name)
+		if err != nil {
+			return fmt.Errorf("failed to create %s: %w", f.name, err)
+		}
+
+		_, err = zf.Write(f.data)
+		if err != nil {
+			return fmt.Errorf("failed to write to %s: %w", f.name, err)
+		}
+	}
+
+	return nil
+}
+
+func getPods(ctx context.Context, namespace string) (podList, error) {
+	// Declare buffer type var for kubectl pipe
+	var podsBuff bytes.Buffer
+
+	// Get all pod names as json
+	getPods := exec.CommandContext(ctx, "kubectl", "-n", namespace, "get", "pods", "-l", "deploy=sourcegraph", "-o=json")
+	getPods.Stdout = &podsBuff
+	getPods.Stderr = os.Stderr
+	err := getPods.Run()
```

Comment:
You're not checking if this error is non-nil.

Reply:
Checked